docs/guides/advanced-routing.md

```diff
@@ -0,0 +1,332 @@
+# Advanced HTTP Routing
+
+In this guide we will be configuring a more advanced set of routing rules with multiple applications. These rules
+will showcase different paths, parameters, and methods used to fine-tune the flow of traffic. For an introduction
+to exposing your application, it is recommended to go through the
+[basic guide](/docs/guides/how-to-direct-traffic-to-your-app.md) first.
+
+The following image shows the traffic flow that we will be creating with these rules.
+
+![Traffic Flow Diagram](/docs/images/advanced-routing.png)
```

Comment:
Would it be better if we changed "Public Endpoint" to "Public IP Address". We wanted to include both hostname and IP in the same box since they are essentially the same point in the traffic flow (thinking of the production case where DNS records exist).

Reply:
This is tricky because the level of abstraction we're hitting here isn't show service resolution either. I think I agree with Saylor on including both though, as technically the call will be to the IP address with cafe.example.com set as the host header.

Reply:
> Would it be better if we changed "Public Endpoint" to "Public IP Address" 

public IP makes sense to me

Reply:
FYI: I changed this to `Public IP Address for cafe.example.com` in the basic use case guide: https://github.com/nginxinc/nginx-kubernetes-gateway/pull/991#discussion_r1300654122